### ISSUE TITLE
Dá uma geral na lib priorizando a memória

### ISSUE THREAD

**Issue by vinebarbosa** [IFRN-robotica-CM/CosmosNV2#12]

**Comment by fecaridade** [comment]
![Captura de tela de 2021-04-03 00-15-51](https://user-images.githubusercontent.com/38886973/113466386-cc6d4480-9411-11eb-8a62-7e3227d33363.png)

Existe a necessidade de todos esses delay?, durante o uso do delay o Arduíno "para", estou considerando e trocando todos os delays possíveis pela função milis()

milis(doc) - https://www.arduino.cc/reference/pt/language/functions/time/millis/
milis(exemplo "paralelismo") - https://www.filipeflop.com/blog/subtituindo-delay-por-millis-no-arduino/

**Comment by fecaridade** [comment]
![Captura de tela de 2021-04-03 00-19-22](https://user-images.githubusercontent.com/38886973/113466440-4271ab80-9412-11eb-9dc2-aed5214b45d9.png)

e isso se repete aqui também, caso seja realmente necessário, abre um PR explicando o pq comentando o código

**Comment by fecaridade** [comment]
![Captura de tela de 2021-04-03 00-20-42](https://user-images.githubusercontent.com/38886973/113466463-72b94a00-9412-11eb-9b44-7a1829c8233a.png)


Ideia de troca do delay para o milis junto dos comentarios

**Comment by fecaridade** [comment]
> ![Captura de tela de 2021-04-03 00-20-42](https://user-images.githubusercontent.com/38886973/113466463-72b94a00-9412-11eb-9b44-7a1829c8233a.png)
> 
> Ideia de troca do delay para o milis junto dos comentarios

Inclusive, cabe a ideia de passar o tempo que o sensor deve bipar por parâmetro

**Comment by fecaridade** [comment]
https://www.youtube.com/watch?v=-XPrSScamXc

da para usar ate os registros como feito em partes do código já, fazer um intermediário do que seria legível para outros também, ou reforçar a documentação.

**Comment by vinebarbosa** [comment]
@fecaridade Sim, é necessários aqueles delays na biblioteca de sensores de cor porque é o tempo que leva para o sensor fazer alguns ciclos de leitura e retornar o resultado. Os delays no biblioteca de sensores de distância também são necessários. Isso porque o sensor precisa receber um pulso de 10 ms para poder ativar o sensor.
